File: persistence/src/main/java/com/example/diploma/persistence/dto/admin/ProductDto.java
package com.example.diploma.persistence.dto.admin;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldDefaults;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class ProductDto {

    Long id;
    String name;
    String brand;
    Double price;
    List<SizeDto> sizes;
    List<ColorDto> colors;
    String imageName;
    CategoryDto category;
}

File: admin/src/main/frontend/src/app/dto/ProductDto.ts
import {CategoryDto} from "./CategoryDto";
import {SizeDto} from "./SizeDto";
import {ColorDto} from "./ColorDto";

export class ProductDto {

	id: number;
	name: string;
	imageName: string;
	brand: string;
	price: number;
	category: CategoryDto;
	sizes: Array<SizeDto>;
	colors: Array<ColorDto>;

	constructor(id?: number, name?: string, imageName?: string, brand?: string, price?: number, category?: CategoryDto,
							sizes?: Array<SizeDto>, colors?: Array<ColorDto>) {
		this.id = id;
		this.name = name;
		this.imageName = imageName;
		this.brand = brand;
		this.price = price;
		this.category = category;
		this.sizes = sizes;
		this.colors = colors;
	}

}
